include/frontend-switcher.php

```diff
@@ -0,0 +1,146 @@
+<?php
+
+
+class PLL_Frontend_Switcher extends PLL_Switcher {
+
+	/**
+	 * Get the language elements for use in a walker
+	 *
+	 * @since 1.2
+	 *
+	 * @param PLL_Frontend_Links $links Instance of PLL_Frontend_Links.
+	 * @param array              $args  Arguments passed to {@see PLL_Switcher::the_languages()}.
+	 * @return array Language switcher elements.
+	 */
+	protected function get_elements( $links, $args ) {
+		$first = true;
+		$out   = array();
+
+		foreach ( $links->model->get_languages_list( array( 'hide_empty' => $args['hide_if_empty'] ) ) as $language ) {
+			list( $id, $order, $slug, $locale, $classes, $url ) = $this->init_foreach_language( $language );
+
+			$curlang = $links->curlang->slug;
+
+			$current_lang = $curlang == $slug;
+
+			if ( $this->manage_url( $classes, $args, $links, $language, $url ) ) {
+				list( $url, $no_translation, $classes ) = $this->manage_url( $classes, $args, $links, $language, $url );
+			} else {
+				continue;
+			}
```

Comment:
If the `PLL_Admin_Switcher` had the `manage_url()` method as well, we could share the whole `get_elements()` method in their parent class.